docs/0.3-pre-upgrade-to-0.3.3.md

```diff
@@ -0,0 +1,93 @@
+
+## 0.3 upgrade steps
+
+Run the 0.3 upgrade helper script. This script will do some basic sanity checks for your ansible inventory, prod-specific.yml and the system's torrc file are properly configured and then run the correct ansible playbooks.
+
+* Run the upgrade helper script
+
+`sudo ./migration_scripts/0.3/upgrade.sh`
+
+## Notes about the 0.3 upgrade helper script
+
+`Error: This script must be run as root.`
+
+* Use the 'sudo' command and provide the administrative password you created at the beginning of the Tails session. i.e. `sudo ./upgrade.sh`
+
+`Error: This script must be run on Tails with a persistent volume.`
+
+* In order for this to work, you must be running Tails on the admin workstation and have access to a persistent volume. Did you unlock your persistent storage at the beginning of the Tails session?
+
+`Error: There is no SSH key file present.`
+
+* When you installed SecureDrop, you generated an SSH key (and probably saved to the default location: `~/.ssh/id_rsa`) for the user 'amnesia' and copied it to both the App and Mon server. You should still have access to that key through the OpenSSH persistence in Tails. If you got this error, check if you lost the key or saved it to a different location. You won't be able to login to the servers remotely without it.
+
+`Error: This script must be run with SecureDrop's git repository cloned to 'securedrop' in your Persistent folder.`
+
+* When you first installed SecureDrop, we assume that you cloned our GitHub repository to a specific location: `/home/amnesia/Persistent/securedrop`. If it's not there, you either need to move it there or start anew.
+
+`Error: There are no HidServAuth values in your torrc file.`
+
+* Did you run the 'SecureDrop Init' script located in the Persistent folder? The init script and the additions it makes to Tor's configuration file are needed in order to access the App and Mon servers using SSH over Tor.
+
+`Error: monitor_ip or app_ip in prod-specific.yml is not an IP address.`
+
+* The production playbook in `/install_files/ansible-base/prod-specific.yml` needs to be fully filled out, with the local IP address for each server specified.
+
+`Error: ssh_users is not defined in prod-specific.yml.`
+
+* Inside the production playbook you must have ssh_users defined as the user which you use to log in to the App and Mon servers, which was created when you installed Ubuntu Server.
+
+`Error: the app or mon ansible_ssh_host in Ansible's inventory file is not an .onion address.`
+
+* Our new provisioner, Ansible, must be run over Tor in order to reach the servers. Replace the IP addresses in `/install_files/ansible-base/inventory` with the .onion hostnames for the App and Mon server's Tor hidden services for SSH.
+
+`Error: can't connect to the Application or Monitor Server via SSH.`
+
+* Something's wrong and we can't connect. You can re-run the script to try again.
+
+** Is the Vidalia connection indicator green?
```

Comment:
This is not the correct syntax for nested bulleted lists in Markdown (it is correct for Creole).